cql3/statements/cf_prop_defs.hh

```diff
@@ -95,6 +95,7 @@ public:
     std::map<sstring, sstring> get_compaction_options() const;
     std::optional<std::map<sstring, sstring>> get_compression_options() const;
     const cdc::options* get_cdc_options(const schema::extensions_map&) const;
+    std::optional<caching_options> get_caching_options() const;
```

Comment:
Maybe it's simpler not to return optional here, a map can be empty. Returning an optional means everyone has to check if it exists or not.

Reply:
If I remember correctly, this function is used in both create table and alter table. In alter table, an empty map is different than not present at all. The empty map clears out the setting while the missing map leaves the setting unchanged.

Reply:
Makes sense!